pkg/localclusterimport/local_cluster_import_operations.go

```diff
@@ -0,0 +1,188 @@
+	clusterDeployment := &hivev1.ClusterDeployment{}
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	err := o.apiReader.Get(o.context, namespacedName, clusterDeployment)
+	if err != nil {
+		return nil, err
+	}
+	return clusterDeployment, nil
+}
+
+func (o *LocalClusterImportOperations) GetAgentClusterInstall(namespace string, name string) (*hiveext.AgentClusterInstall, error) {
+	agentClusterInstall := &hiveext.AgentClusterInstall{}
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	err := o.apiReader.Get(o.context, namespacedName, agentClusterInstall)
+	if err != nil {
+		return nil, err
+	}
+	return agentClusterInstall, nil
+}
+
+func (o *LocalClusterImportOperations) CreateAgentClusterInstall(agentClusterInstall *hiveext.AgentClusterInstall) error {
+	return o.cachedApiClient.Create(o.context, agentClusterInstall)
+}
+
+func (o *LocalClusterImportOperations) GetNamespace(name string) (*v1.Namespace, error) {
+	ns := &v1.Namespace{}
+	namespacedName := types.NamespacedName{
+		Namespace: "",
+		Name:      name,
+	}
+	err := o.apiReader.Get(o.context, namespacedName, ns)
+	if err != nil {
+		return nil, err
+	}
+	return ns, nil
+}
+
+func (o *LocalClusterImportOperations) GetSecret(namespace string, name string) (*v1.Secret, error) {
+	secret := &v1.Secret{}
+	namespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      name,
+	}
+	err := o.apiReader.Get(o.context, namespacedName, secret)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to fetch secret %s from namespace %s", name, namespace)
+	}
+	return secret, nil
+}
+
+func (o *LocalClusterImportOperations) CreateSecret(namespace string, secret *v1.Secret) (*v1.Secret, error) {
+	err := o.cachedApiClient.Create(o.context, secret)
+	if err != nil {
+		return nil, err
+	}
+	return secret, nil
+}
+
+func (o *LocalClusterImportOperations) GetClusterVersion(name string) (*configv1.ClusterVersion, error) {
+	clusterVersion := &configv1.ClusterVersion{}
+	namespacedName := types.NamespacedName{
+		Namespace: "",
+		Name:      name,
+	}
+	err := o.apiReader.Get(o.context, namespacedName, clusterVersion)
+	if err != nil {
+		return nil, err
+	}
+	return clusterVersion, nil
+}
+
+func (o *LocalClusterImportOperations) GetClusterImageSet(name string) (*hivev1.ClusterImageSet, error) {
+	clusterImageSet := &hivev1.ClusterImageSet{}
+	namespacedName := types.NamespacedName{
+		Namespace: "",
+		Name:      name,
+	}
+	err := o.apiReader.Get(o.context, namespacedName, clusterImageSet)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to fetch cluster image set %s", name)
+	}
+	return clusterImageSet, nil
+}
+
+func (o *LocalClusterImportOperations) CreateClusterImageSet(clusterImageSet *hivev1.ClusterImageSet) (*hivev1.ClusterImageSet, error) {
+	err := o.cachedApiClient.Create(o.context, clusterImageSet)
```

Comment:
what will happen is it's already exists? 
same question to all the resources.

Reply:
It would fail to write the cluster image set and the old one would remain in place. 
Not sure what the best thing to do is, on one hand, we could delete anything we find when performing this write, but seems to me that this would not be the smart choice until we have looked into the impact of the update ticket.

Reply:
until we handle upgrade it should be good enough